packages/patterns/src/getOpaquePattern.ts

```diff
@@ -29,8 +34,11 @@ function getOpaquePattern(pattern) {
 
 export default getOpaquePattern;
 
-function objectWithout(obj, propertiesToRemove) {
-  return Object.keys(obj)
-    .filter((key) => !propertiesToRemove.includes(key))
-    .reduce((newObj, key) => ({ ...newObj, [key]: obj[key] }), {});
+function objectWithout(
+  obj: Record<string, unknown>,
+  propertiesToRemove: Array<string>
```

Comment:
<!-- ideas that popped up from reviewing -->
**💡 thought:**
Should we try to use the same convention across the files for consistency, .e.g always either `Array<string>` or `string[]`? Or no preference? Not sure if this is usually considered important in TS projects. Thoughts?

Reply:
No strong preference, but we could add a lint rule for this: https://typescript-eslint.io/rules/array-type
I think we've mostly used `string[]` so far, also in the PHP code base

Reply:
Yeah, let's go for the more compact syntax.